impl-tinkerpop/src/main/java/org/hawkular/inventory/impl/tinkerpop/AbstractSourcedGraphService.java

```diff
@@ -78,6 +93,121 @@ public Single create(Blueprint blueprint) {
         return createSingleBrowser(FilterApplicator.fromPath(path).get());
     }
 
+    public final void update(E entity) {
+        Constants.Type type = Constants.Type.of(entity);
+        List<String> mappedProperties = Arrays.asList(type.getMappedProperties());
+
+        entity.getProperties().keySet().forEach(k -> {
+            if (mappedProperties.contains(k)) {
+                throw new IllegalArgumentException("Property '" + k + "' is reserved. Cannot set it to a custom value");
+            }
+        });
+
+        Vertex vertex = convert(entity);
```

Comment:
Well, the example you gave is pretty much exactly what `convert(Entity)` does so I think we're cool here.

Btw. IMHO it is always needed to use both `type` and `uid` constraints on `source()` because the `uid` is user provided and I can easily imagine two entities of different types having the same uid. Of course, `source()` puts context (tenant, env, ...) into the search so the probability of clash is lowered by that but still.